src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -56,67 +56,72 @@
+DWORD                      touchIDs[maxTouchPoints];
+POINT                      touches[maxTouchPoints];
 
-const GUID GUID_DEVINTERFACE_HID = {0x4d1e55b2, 0xf16f, 0x11cf, {0x88, 0xcb, 0x00, 0x11, 0x11, 0x00, 0x00, 0x30}};
-
-void setProcessDpiAware()
+// Convert a hardware dependent ID to a 0 based index we can use
+int getTouchID(DWORD id)
 {
-    // Try SetProcessDpiAwareness first
-    HINSTANCE shCoreDll = LoadLibrary(L"Shcore.dll");
-
-    if (shCoreDll)
+    for (int i = 0; i < maxTouchPoints; ++i)
     {
-        enum ProcessDpiAwareness
-        {
-            ProcessDpiUnaware         = 0,
-            ProcessSystemDpiAware     = 1,
-            ProcessPerMonitorDpiAware = 2
-        };
-
-        using SetProcessDpiAwarenessFuncType = HRESULT(WINAPI*)(ProcessDpiAwareness);
-        auto SetProcessDpiAwarenessFunc      = reinterpret_cast<SetProcessDpiAwarenessFuncType>(
-            reinterpret_cast<void*>(GetProcAddress(shCoreDll, "SetProcessDpiAwareness")));
-
-        if (SetProcessDpiAwarenessFunc)
+        if (touchIDs[i] == id)
+            return i;
+    }
+    for (int i = 0; i < maxTouchPoints; ++i)
+    {
+        if (touchIDs[i] == static_cast<DWORD>(-1))
         {
-            // We only check for E_INVALIDARG because we would get
-            // E_ACCESSDENIED if the DPI was already set previously
-            // and S_OK means the call was successful
-            if (SetProcessDpiAwarenessFunc(ProcessSystemDpiAware) == E_INVALIDARG)
-            {
-                sf::err() << "Failed to set process DPI awareness" << std::endl;
-            }
-            else
-            {
-                FreeLibrary(shCoreDll);
-                return;
-            }
+            touchIDs[i] = id;
+            return i;
         }
-
-        FreeLibrary(shCoreDll);
     }
+    return -1;
+}
 
-    // Fall back to SetProcessDPIAware if SetProcessDpiAwareness
-    // is not available on this system
-    HINSTANCE user32Dll = LoadLibrary(L"user32.dll");
+// Get a system error string from an error code
+std::string getErrorString(DWORD error)
+{
+    PTCHAR buffer;
+
+    if (FormatMessage(FORMAT_MESSAGE_MAX_WIDTH_MASK | FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM,
+                      nullptr,
+                      error,
+                      0,
+                      reinterpret_cast<PTCHAR>(&buffer),
+                      0,
+                      nullptr) == 0)
+        return "Unknown error.";
+
+    const sf::String message = buffer;
+    LocalFree(buffer);
+    return message.toAnsiString();
+}
 
-    if (user32Dll)
-    {
-        using SetProcessDPIAwareFuncType = BOOL(WINAPI*)();
-        auto SetProcessDPIAwareFunc      = reinterpret_cast<SetProcessDPIAwareFuncType>(
-            reinterpret_cast<void*>(GetProcAddress(user32Dll, "SetProcessDPIAware")));
+const GUID GUID_DEVINTERFACE_HID = {0x4d1e55b2, 0xf16f, 0x11cf, {0x88, 0xcb, 0x00, 0x11, 0x11, 0x00, 0x00, 0x30}};
```

Comment:
I know this existed before your PR but can we move this constant into the one function out of the global scope and into the 1 function where it's actually being used?

Reply:
To be fair I'm not even 100% sure why it exists as a kind of magic value and isn't in a header. Or maybe it originally wasn't in common Windows headers?

Reply:
It makes sense that it's not in a header since it's only used in one place in this single .cpp file. It should just be a `static const` in the one function that needs it.